internal/testutil/testutil.go

```diff
@@ -94,6 +96,61 @@ func AssertPkgEqual(t *testing.T, sourceDir, destDir string) bool {
 	return assert.Empty(t, diff.List())
 }
 
+// KptfileAwarePkgEqual compares two packages (including any subpackages)
+// and has special handling of Kptfiles to handle fields that contain
+// values which cannot easily be specified in the golden package.
+func KptfileAwarePkgEqual(t *testing.T, pkg1, pkg2 string) bool {
```

Comment:
Do we need an option to just compare parent packages without comparing subpackages ?

Reply:
I haven't found a need for it yet. But I think we might need something in this area eventually, in particular for detecting changes to packages during updates. If this is needed, I will try to reconcile the logic in one place.